buildconfig/CMake/FindPyQt.py

```diff
@@ -8,68 +8,72 @@
-              possible_sip_dirs = []
-              if qt_maj_version == '4':
-                  possible_sip_dirs.append(os.path.join('pyqt@4', 'share', 'sip'))
-                  possible_sip_dirs.append(os.path.join('mantid-pyqt@4', 'share', 'sip'))
-              elif qt_maj_version == '5':
-                  possible_sip_dirs.append(os.path.join('pyqt', 'share', 'sip', 'Qt5'))
-                  possible_sip_dirs.append(os.path.join('mantid-pyqt5', 'share', 'sip', 'Qt5'))
-              else:
-                  raise RuntimeError("Unknown Qt version ({}) found. Unable to determine location of PyQt sip files."
-                                     "Please update FindPyQt accordingly.".format(self.version_str[0]))
-          else:
-              prefix = os.path.join(sys.prefix, 'share')
-              possible_sip_dirs = (
-                  'python{}{}-sip/{}'.format(sys.version_info.major, sys.version_info.minor, name),
-                  'python{}-sip/{}'.format(sys.version_info.major, name),
-                  'sip/{}'.format(name)
-              )
-          for sip_dir in possible_sip_dirs:
-              pyqt_sip_dir = os.path.join(prefix, sip_dir)
-              if os.path.exists(pyqt_sip_dir):
-                  self.sip_dir = pyqt_sip_dir
-                  break
-          if self.sip_dir is None:
-              possible_sip_dirs = list(map(lambda p: os.path.join(prefix, p), possible_sip_dirs))
-              raise RuntimeError("Unable to find {}. "
-                                 "Tried following locations: {}".format(name, pprint.pformat(possible_sip_dirs)))
-
-      # Assume uic script is in uic submodule
-      uic = __import__(name + '.uic', globals(), locals(), ['uic'], 0)
-      self.pyuic_path = os.path.join(os.path.dirname(uic.__file__), 'pyuic.py')
+class PyQtConfig:
+    """Inspects the installed PyQt version and extracts
+    information about it.
+    """
+
+    version_hex: str
+    version_str: str
+    qt_tag: str
+    sip_dir: str
+    sip_flags: str
+    pyuic_path: str
+
+    def __init__(self, major_version):
+        """Inspects the named PyQt package given the major version number
+        :param major_version: The major version number of the library
+        """
+        pyqt_name = f'PyQt{major_version}'
+        qtcore = __import__(f'{pyqt_name}.QtCore', globals(), locals(), ['QtCore'], 0)
+        self.version_hex = qtcore.PYQT_VERSION
+        self.version_str = qtcore.PYQT_VERSION_STR
+        self.sip_flags = qtcore.PYQT_CONFIGURATION['sip_flags']
+        self.qt_tag = self._get_qt_tag(self.sip_flags)
+        # This is based on QScintilla's configure.py, and only works for the
+        # default case where installation paths have not been changed in PyQt's
+        # configuration process.
+        if sys.platform == 'win32':
+            if sys.version_info > (3, ):
+                self.sip_dir = os.path.join(sys.prefix, 'share', 'sip', pyqt_name)
+            else:
```

Comment:
This Python 2 branch is dead, we might as well tidy it up whilst were here as I suspect it will be forgotten about in the future